src/main/java/org/folio/circulation/domain/CreateRequestService.java

```diff
@@ -24,16 +28,23 @@
   private final RequestLoanValidator requestLoanValidator;
   private final RequestNoticeSender requestNoticeSender;
   private final UserManualBlocksValidator userManualBlocksValidator;
+  private final EventPublisher eventPublisher;
 
   public CreateRequestService(CreateRequestRepositories repositories,
-    UpdateUponRequest updateUponRequest, RequestLoanValidator requestLoanValidator,
-    RequestNoticeSender requestNoticeSender, UserManualBlocksValidator userManualBlocksValidator) {
-
+                              UpdateUponRequest updateUponRequest, RequestLoanValidator requestLoanValidator,
+                              RequestNoticeSender requestNoticeSender, UserManualBlocksValidator userManualBlocksValidator, EventPublisher eventPublisher) {
     this.repositories = repositories;
     this.updateUponRequest = updateUponRequest;
     this.requestLoanValidator = requestLoanValidator;
     this.requestNoticeSender = requestNoticeSender;
     this.userManualBlocksValidator = userManualBlocksValidator;
+    this.eventPublisher = eventPublisher;
+  }
+
+  public CreateRequestService(CreateRequestRepositories repositories,
+    UpdateUponRequest updateUponRequest, RequestLoanValidator requestLoanValidator,
+    RequestNoticeSender requestNoticeSender, UserManualBlocksValidator userManualBlocksValidator) {
+    this(repositories, updateUponRequest, requestLoanValidator, requestNoticeSender, userManualBlocksValidator, null);
```

Comment:
This is quite error prone, I think. Could you please remove this constructor and pass the non-null event publisher?

Alternatively you can create a factory method, e.g.:
```java
public static CreateRequestService using(Clients clients) {
    final LoanRepository loanRepository = new LoanRepository(clients);
    final LoanPolicyRepository loanPolicyRepository = new LoanPolicyRepository(clients);
    final RequestNoticeSender requestNoticeSender = RequestNoticeSender.using(clients);
    final ConfigurationRepository configurationRepository = new ConfigurationRepository(clients);

    final var repositories = new CreateRequestRepositories(RequestRepository.using(clients),
      new RequestPolicyRepository(clients), configurationRepository,
      new AutomatedPatronBlocksRepository(clients));

    final FindWithCqlQuery<UserManualBlock> userManualBlocksValidator
      = findWithCqlQuery(clients.userManualBlocksStorageClient(),
      "manualblocks", UserManualBlock::from);

    final UpdateUponRequest updateUponRequest = new UpdateUponRequest(
      new UpdateItem(clients),
      new UpdateLoan(clients, loanRepository, loanPolicyRepository),
      UpdateRequestQueue.using(clients));

    return new CreateRequestService(repositories,
      updateUponRequest, new RequestLoanValidator(loanRepository),
      requestNoticeSender, new UserManualBlocksValidator(userManualBlocksValidator),
      new EventPublisher(clients.pubSubPublishingService()));
}
```

And use the method instead of constructor.